File: PATTERNS/Pattern46/Pattern46.py
rows = int(input("Enter the number of rows in the butterfly pattern: "))

# Print upper part of the butterfly
for i in range(1, rows + 1):
    for j in range(1, i + 1):
        print(j, end=" ")
    for j in range(1, 2 * (rows - i) + 1):
        print(" ", end=" ")
    for j in range(i, 0, -1):
        print(j, end=" ")
    print()


for i in range(1, rows + 1):
    for j in range(1, i + 1):
        print(" ", end=" ")
    for j in range(rows, i - 1, -1):
        print(j, end=" ")
    for j in range(rows - i, 0, -1):
        print(j, end=" ")
    print()

File: PATTERNS/Pattern46/Pattern46.c
#include <stdio.h>

int main() {
    int rows;
    printf("Enter the number of rows: ");
    scanf("%d", &rows); // Number of rows
    
    // Print the upper half of the butterfly pattern
    for (int i = 1; i <= rows; i++) {
        // Print numbers in increasing order
        for (int j = 1; j <= i; j++) {
            printf("%d ", j);
        }
        
        // Print spaces
        for (int j = 1; j <= 2 * (rows - i); j++) {
            printf("  ");
        }
        
        // Print numbers in decreasing order
        for (int j = i; j >= 1; j--) {
            printf("%d ", j);
        }
        
        printf("\n");
    }

    // Print the lower half of the butterfly pattern
    for (int i = 1; i <= rows; i++) {
        // Print spaces
        for (int j = 1; j <= i; j++) {
            printf("  ");
        }
        
        // Print numbers in decreasing order
        for (int j = rows; j >= i; j--) {
            printf("%d ", j);
        }
        
        // Print numbers in decreasing order
        for (int j = rows - i; j >= 1; j--) {
            printf("%d ", j);
        }
        
        printf("\n");
    }

    return 0;
}
